server/config/config.go

```diff
@@ -665,9 +690,53 @@ func (c *ScheduleConfig) adjust(meta *configMetaData) error {
 	adjustFloat64(&c.HighSpaceRatio, defaultHighSpaceRatio)
 	adjustSchedulers(&c.Schedulers, defaultSchedulers)
 
+	for k, b := range c.migrateConfigurationMap() {
+		if err := c.parseDeprecatedFlag(meta, k, b[0], b[1]); err != nil {
+			return err
+		}
+	}
+
 	return c.Validate()
 }
 
+func (c *ScheduleConfig) migrateConfigurationMap() map[string][2]*bool {
+	return map[string][2]*bool{
+		"remove-down-replica":     {&c.DisableRemoveDownReplica, &c.EnableRemoveDownReplica},
+		"replace-offline-replica": {&c.DisableReplaceOfflineReplica, &c.EnableReplaceOfflineReplica},
+		"make-up-replica":         {&c.DisableMakeUpReplica, &c.EnableMakeUpReplica},
+		"remove-extra-replica":    {&c.DisableRemoveExtraReplica, &c.EnableRemoveExtraReplica},
+		"location-replacement":    {&c.DisableLocationReplacement, &c.EnableLocationReplacement},
+		"namespace-relocation":    {&c.DisableNamespaceRelocation, &c.EnableNamespaceRelocation},
+	}
+}
+
+func (c *ScheduleConfig) parseDeprecatedFlag(meta *configMetaData, name string, old, new *bool) error {
+	oldName, newName := "disable-"+name, "enable-"+name
+	defineOld, defineNew := meta.IsDefined(oldName), meta.IsDefined(newName)
+	if defineNew && defineOld {
+		if *new == *old {
+			return errors.Errorf("config item %s and %s(deprecated) are conflict", newName, oldName)
+		}
+	} else if defineOld {
+		*new = !*old // only define disable-*, use !disable-*.
+	} else if !defineNew {
+		*new = true // default value is true
+	}
+	*old = false
+	return nil
+}
+
+// MigrateDeprecatedFlags updates new flags according to deprecated flags.
+func (c *ScheduleConfig) MigrateDeprecatedFlags() {
+	c.DisableLearner = false
+	for _, b := range c.migrateConfigurationMap() {
+		// If old=false (previously disabled), set both old and new to false.
```

Comment:
why?

Reply:
`new` should be false because it is disabled previously, and `old` should be reset to `false` to make it be ignored when marshal the config to json.